### DISCUSSION TITLE
Unsure about phase portraits

### DISCUSSION THREAD

**Discussion by benjytan88** [theislab/scvelo#503] [Q&A]
Hi all, 
This is my first time using scVelo. I am slightly loss with the interpretation of my phase portraits and whether my data is suitable for velocity analysis. Hope to get some help here. 

Briefly, I am analyzing CD4 T-cells from healthy individuals (n=3) and cancer patients (n=2). I have around 8,200 cells and ran them with the default steps as described in the tutorial. When estimating the velocity, I used the `dynamical` mode. 

The results from my data are shown below. 

![UMAP clustering](https://user-images.githubusercontent.com/24598265/122531214-42209e00-d05a-11eb-8cd5-cc2fd03e8dc9.png)

The bottom left cluster (1,4,6) are mainly cells from healthy individuals while the remaining clusters are made up of cells from cancer patients. 

![Velocity embeddings / stream](https://user-images.githubusercontent.com/24598265/122532149-2e296c00-d05b-11eb-9117-00cf4524b1a1.png)

![Transcript proportions](https://user-images.githubusercontent.com/24598265/122532165-32ee2000-d05b-11eb-8b04-d41b65512fd3.png)

**Velocity plots of top 4 variable genes**
![Velocity plots of top 4 variable genes](https://user-images.githubusercontent.com/24598265/122532166-32ee2000-d05b-11eb-9d22-288eeb868175.png)

**Phase portraits of top 20 likelihood genes**
![Phase portraits of top 20 likelihood genes](https://user-images.githubusercontent.com/24598265/122532170-3386b680-d05b-11eb-90ff-ffef24f06e20.png)

My questions are as follows:

1. During selection of variable genes for velocity estimation, my final list only contains 6 genes after pre-processing. That is too low, right? Then, how accurate is my velocity estimation? Because according to the vectors, the direction is as what we expected biologically. 
2. The proportion of unspliced transcripts in our dataset is too high. I don't know why but the only reason I can think of is that we used a 5'-end targeting for mRNA capture (10x Chromium Immune Profiling kit). Could that be the reason and how will that affect the velocity estimation? 
3. I'm not sure if I understand correctly, but from the phase portraits, I think that probably my data is not really suitable for velocity analysis. Is that so? From the velocity plot, there's no significant contribution of a gene to any phase and all my phase portraits looks like they are in a steady-state. 
4. Going through issues & discussions, I saw mentions that a good portrait should have a football / almond-shape. What do you mean by that? I don't really get it. I have gone through the tutorial and the GIF image. Which of these below is a good phase portrait? I was thinking that the first set would be the good ones. 

**First set - good phase portraits??** 
![First set - good phase portraits?](https://user-images.githubusercontent.com/24598265/122533768-e0adfe80-d05c-11eb-8830-a3728a189221.png)

**Second set - are these what is called football / almond-shape?** 
![Second set](https://user-images.githubusercontent.com/24598265/122533785-e3105880-d05c-11eb-8fde-d4773df8cad7.png)

Velocity analysis is really interesting and I hope to be able to use it for my dataset. 
Feedback on my questions and dataset will be highly appreciated. 
Thank you very much! (^_^)

**Comment by WeilerP** [comment]
@nicodemus88, let me try to answer (at least some of) your questions.

> 1. During selection of variable genes for velocity estimation, my final list only contains 6 genes after pre-processing. That is too low, right? Then, how accurate is my velocity estimation? Because according to the vectors, the direction is as what we expected biologically.

Yes, 6 seems low. Did you check why most of your genes are being filtered out? If only 6 genes are kept, how can you show the phase portraits for at least 20 genes? Concerning the directionality: I personally believe this is pure luck based on the phase portraits you are showing.

> 2. The proportion of unspliced transcripts in our dataset is too high. I don't know why but the only reason I can think of is that we used a 5'-end targeting for mRNA capture (10x Chromium Immune Profiling kit). Could that be the reason and how will that affect the velocity estimation?

Sorry, not sure if this is a problem. @VolkerBergen, do you have any helpful insight concerning this?

> 3. I'm not sure if I understand correctly, but from the phase portraits, I think that probably my data is not really suitable for velocity analysis. Is that so? From the velocity plot, there's no significant contribution of a gene to any phase and all my phase portraits looks like they are in a steady-state.

Yes, I'd also say that RNA velocity fails here. Your phase portraits do not show any curvature. This is either do to genes being in steady state or the data being too noisy.

> 4. Going through issues & discussions, I saw mentions that a good portrait should have a football / almond-shape. What do you mean by that? I don't really get it. I have gone through the tutorial and the GIF image. Which of these below is a good phase portrait? I was thinking that the first set would be the good ones.

Yes, in the ideal phase portrait, we observe (part of) a football shape. In the supplementary material of [Generalizing RNA velocity to transient cell states through dynamical modeling](https://www.nature.com/articles/s41587-020-0591-3), you'll find phase portraits of simulated data:

<img width="403" alt="ideal_phase_portraits" src="https://user-images.githubusercontent.com/28675704/122545847-9c0b6e80-d02e-11eb-8d2e-8464ea6202a8.png">

Here, the data forms a football/almond. The curvature comes from a temporal delay during splicing dynamics. For constant rates, the upper arc corresponds to induction (unspliced RNA is produced prior to spliced RNA, i.e. its abundance increases first). Similarly, the lower arc is the repression phase.
The phase portrait of *Cpe* is a good phase portrait. You can imagine it as part of a football. In biological terms: Assuming constant rates, this is the induction phase. *Adk* is still okay, but the curvature is not as predominant as for the previous one.  In case of *HMGB2*, you cannot observe a nice cell state transition and the fit is dominated by the brown cluster. *Col13a1* could still be okay, although you do not observe a highly resoluted evolution as for *Cpe*, for example.
Hope this helps/answers your question.

**Reply by benjytan88** [reply]
@WeilerP Thank you very much for your reply. 

> Yes, 6 seems low. Did you check why most of your genes are being filtered out? If only 6 genes are kept, how can you show the phase portraits for at least 20 genes? Concerning the directionality: I personally believe this is pure luck based on the phase portraits you are showing.

I am not sure how to check. Could you please guide me on how can I check why the genes are filtered out? 
And the 20 phase portraits are the top 20 likelihood genes (driver genes) obtained through dynamical analysis. 

> The proportion of unspliced transcripts in our dataset is too high. I don't know why but the only reason I can think of is that we used a 5'-end targeting for mRNA capture (10x Chromium Immune Profiling kit). Could that be the reason and how will that affect the velocity estimation?

> Sorry, not sure if this is a problem. @VolkerBergen, do you have any helpful insight concerning this?

Roughly, we expect to see approximately 15 - 20% reads mapping to the intron in circulating PBMCs, but our mapping rate was way too high. I was wondering if the pre-processing to obtain the loom file with velocyto is not working really well with 5'-end protocols. 

> Hope this helps/answers your question.

Yes, thank you very much. Your explanation makes it clearer and now I am able to understand the phase portraits better. 

Thank you very much for your response! 
Hope to get some feedback soon on the compatibility of scVelo with 5'-end protocols.

**Comment by erzakiev** [comment]
I also obtain a very low splicing rate (~6% overall). This results in the downstream failure of velocity computation. Where would you recommend to start digging in? 
```python
computing velocities
WARNING: You seem to have very low signal in splicing dynamics.
The correlation threshold has been reduced to 0.0.
Please be cautious when interpreting results.
[/Users/administrateur/miniforge3/envs/scFates/lib/python3.11/site-packages/scvelo/tools/optimization.py:184](http://localhost:8888/Users/administrateur/miniforge3/envs/scFates/lib/python3.11/site-packages/scvelo/tools/optimization.py#line=183): DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  gamma[i] = np.linalg.pinv(A.T.dot(A)).dot(A.T.dot(y[:, i]))

```
![scvelo_proportions__splicing_rate](https://github.com/theislab/scvelo/assets/13952639/23a6414d-e0f4-4356-a2fe-cdf66f20b2fb)
